ktoml-core/src/commonMain/kotlin/com/akuleshov7/ktoml/parsers/TomlParser.kt

```diff
@@ -40,18 +40,34 @@ public value class TomlParser(private val config: TomlConfig) {
         // here we always store the bucket of the latest created array of tables
         var latestCreatedBucket: TomlArrayOfTablesElement? = null
 
+        val comments: MutableList<String> = mutableListOf()
+
         mutableTomlLines.forEachIndexed { index, line ->
             val lineNo = index + 1
             // comments and empty lines can easily be ignored in the TomlTree, but we cannot filter them out in mutableTomlLines
             // because we need to calculate and save lineNo
-            if (!line.isComment() && !line.isEmptyLine()) {
+            if (line.isComment()) {
+                if (!config.ignoreComments) {
+                    comments += line.trimCommentLine()
+                }
+            } else if (!line.isEmptyLine()) {
+                // Parse the inline comment if any
+                val inlineComment = if (config.ignoreComments) {
+                    ""
+                } else {
+                    when (val hashIndex = line.indexOf('#')) {
+                        -1 -> ""
+                        else -> line.substring(hashIndex + 1).trim()
+                    }
+                }
+
                 if (line.isTableNode()) {
                     if (line.isArrayOfTables()) {
                         // TomlArrayOfTables contains all information about the ArrayOfTables ([[array of tables]])
                         val tableArray = TomlArrayOfTables(line, lineNo, config)
                         val arrayOfTables = tomlFileHead.insertTableToTree(tableArray, latestCreatedBucket)
                         // creating a new empty element that will be used as an element in array and the parent for next key-value records
-                        val newArrayElement = TomlArrayOfTablesElement(lineNo, config)
+                        val newArrayElement = TomlArrayOfTablesElement(lineNo, comments.toList(), inlineComment, config)
```

Comment:
May be accept Mutable list in the Toml nodes? Instead of making `toList` everywhere?
Or not... As you wish 😄

Reply:
Well, `toList` is there to copy the elements to a readonly list. If we passed the `comments` list itself the comments would be modified when the next element is being parsed

Maybe it's better to create a new list every element instead